var/spack/repos/builtin/packages/json-c/package.py

```diff
@@ -16,6 +16,7 @@ class JsonC(AutotoolsPackage):
     version('0.11',   'aa02367d2f7a830bf1e3376f77881e98')
 
     depends_on('autoconf', type='build')
+    depends_on('perl@5.16.3')
```

Comment:
This is too specific -- it'll force every thing that builds with `json-c` to use Perl 5.16.3.  If this is the minimum version needed it should say `5.16.3:`

I'm also nearly positive this is a build dependency inherited through `autoconf`, so it needs to be specified (by default it's build/link and will add RPATHs for perl libs)

Reply:
Updated to be a minimum version of 5.16.3:. I can't comment on if it's autoconf related, so I'll leave it here for now for @hartzell to comment.

Reply:
Autoconf/*etc...* require perl when they run.  I haven't found a firm statement of what version they require, the best is [this autotools announcement from 2012](https://lists.gnu.org/archive/html/autotools-announce/2012-04/msg00004.html) that mentions that autotools now require **perl 5.6 or better**.

Ah, [here's a bit from autoconf's *own* `configure.ac`](https://git.savannah.gnu.org/gitweb/?p=autoconf.git;a=blob;f=configure.ac;h=0f8c006e10e5946524a8a787df4d88186273f3a8;hb=HEAD#l140), so 5.6 still looks right (keeping in mind perl's internal [5.006] vs external [5.6] version representation...).

```sh
140 $PERL -e 'require 5.006;' || {
141    AC_MSG_ERROR([Perl 5.006 or better is required])
142 }
```

I can't replicate the failure on my CentOS boxes, but I think that's because they have a decent enough `/usr/bin/perl`.  The module that @ziya5635 had trouble with is `strict.pm` which is `/usr/share/perl5/strict.pm` for the CentOS system perl.  It's a fundamental bit of the perl core, I can't imagine it going missing....  I've asked @ziya5635 for additional info about how he was triggering his issue.

I'd be inclined to do `depends_on('perl', type='run')` and get fancier when there's data.  If we ever add a perl *older* than 5.6 to the tree, we have other problems... :scream:

I'll follow up if I can recreate the bug in #11613.

Reply:
Thanks @hartzell you're awesome! I too wasn't able to reproduce the issue, I'm not on Ubuntu 18.04 (I upgraded!)

Reply:
An edit, to my earlier comment, I'd be inclined to add a `depends_on('perl', type='run')` **to autoconf**.